state/store.go

```diff
@@ -73,6 +79,20 @@ type Store interface {
 	Bootstrap(State) error
 	// PruneStates takes the height from which to start pruning and which height stop at
 	PruneStates(int64, int64, int64) error
+	// PruneABCIResponses will prune all ABCI responses below the given height.
+	PruneABCIResponses(int64) (uint64, error)
+	// SaveApplicationRetainHeight persists the application retain height from the application
+	SaveApplicationRetainHeight(height int64) error
+	// GetApplicationRetainHeight returns the retain height set by the application
+	GetApplicationRetainHeight() (int64, error)
+	// SaveDataCompanionRetainHeight saves the retain height set by the data companion
+	SaveDataCompanionRetainHeight(height int64) error
+	// GetDataCompanionRetainHeight returns the retain height set by the data companion
+	GetDataCompanionRetainHeight() (int64, error)
```

Comment:
Similarly to the previous comment, should probably be called something like `GetCompanionBlockRetainHeight`.